homeassistant/components/binary_sensor/axis.py

```diff
@@ -23,45 +23,66 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
     add_entities([AxisBinarySensor(hass, discovery_info)], True)
 
 
-class AxisBinarySensor(AxisDeviceEvent, BinarySensorDevice):
+class AxisBinarySensor(BinarySensorDevice):
     """Representation of a binary Axis event."""
 
     def __init__(self, hass, event_config):
         """Initialize the Axis binary sensor."""
         self.hass = hass
-        self._state = False
-        self._delay = event_config[CONF_TRIGGER_TIME]
+        self.axis_event = event_config[CONF_EVENT]
+        self.device_name = event_config[CONF_NAME]
+        self.delay = event_config[CONF_TRIGGER_TIME]
+        self.location = event_config[ATTR_LOCATION]
         self._timer = None
-        AxisDeviceEvent.__init__(self, event_config)
-
-    @property
-    def is_on(self):
-        """Return true if event is active."""
-        return self._state
-
-    def update(self):
-        """Get the latest data and update the state."""
-        self._state = self.axis_event.is_tripped
+        self.axis_event.callback = self._update_callback
 
     def _update_callback(self):
         """Update the sensor's state, if needed."""
-        self.update()
-
         if self._timer is not None:
             self._timer()
```

Comment:
Maybe change name of this attribute to clarify that it holds the function that unsubscribes the event listener.

Reply:
`remove_timer` perhaps? It's weird to call it `timer` when calling the function cancels the timing action.

Reply:
Let's go with that!